recursion/countUnorderedAdditionOfSum.py

```diff
@@ -33,3 +33,23 @@
 #     4 = 2 + 2
 #     4 = 3 + 1
 #     4 = 4
+def countUnorderedAdditionOfSum(sum: int) -> int:
+    if sum < 0:
+        return 0
+
+    memo = [None] * (sum + 1)
+    memo[0] = 1
+
+    def countUnorderedAdditionOfSumHelper(sum: int) -> int:
+        if memo[sum] is not None:
```

Comment:
I told you that if memo[sum] is None, then memo[sum] is False otherwise it is True. You don’t need to check if it is not None. Just:

    if memo[sum]:
        <code block>